packages/lodestar/src/sync/regular/naive/naive.ts

```diff
@@ -154,12 +171,41 @@ export class NaiveRegularSync implements IRegularSync {
   }
 
   private getSyncPeers = async (): Promise<PeerId[]> => {
-    return this.network.getPeers().reduce( (validPeers: PeerId[], peer: PeerId) => {
-      const rep = this.reps.getFromPeerId(peer);
-      if(rep && rep.latestStatus) {
-        validPeers.push(peer);
+    if (!this.network.getPeers().includes(this.bestPeer)) {
+      // bestPeer disconnected
+      this.bestPeer = undefined;
+      await this.waitForBestPeer();
+    }
+    return [this.bestPeer];
+  };
+
+  private waitForBestPeer = async (): Promise<void> => {
+    // check every slot
```

Comment:
Can we only send a status if our head slot is < getCurrentSlot to avoid spamming peers?

Reply:
I checked that in start() so if we detect genesis at the same time with other peer we don't call `waitForBestPeer` and finish RegularSync immediately